core/src/main/java/ch/cyberduck/core/Profile.java

```diff
@@ -536,6 +536,57 @@ public boolean isOAuthPKCE() {
         return this.bool("OAuth PKCE");
     }
 
+    @Override
```

Comment:
As previously discussed in https://github.com/iterate-ch/cyberduck/pull/14796/files/36d33a7a4e0b2aeb52b33ae3b762c75cff23e338#r1229312937 please use custom properties instead in the profile and set defaults if required in `defaults.properties`.

Reply:
Good idea. It is to clarify where the definition of the s3.bucket.virtualhost.disable is? @chenkins used it in the Properties section also. 
<key>Properties</key>
        <array>
            <string>s3.bucket.virtualhost.disable=true</string>
        </array>
        
In my opinion this works only if we can garantee that the Properties are only used for the request parameters:
//                .withRoleArn("consoleAdmin")
//                .withRoleSessionName("testSession");
//                .withDurationSeconds(900);

Reply:
A default definition can be added in `default.properties` [^1] if applicable and provided in the *.cyberduckprofile files for specific services.

[^1]: https://github.com/iterate-ch/cyberduck/blob/master/defaults/src/main/resources/default.properties

Reply:
Thanks. We set the default.properties. We don't know how to overwrite the default values by the custom properties.
if (host.getProtocol().getProperties().containsKey("s3.assumerole.durationseconds")) {
            System.out.println("set duration second with profile value");
            webIdReq = webIdReq.withDurationSeconds(host.getProtocol().getProperties().get("s3.assumerole.durationseconds"));
        }
        else {
            webIdReq = webIdReq.withDurationSeconds(new HostPreferences(host).getInteger("s3.assumerole.durationseconds"));
        }

Reply:
Just use `new HostPreferences(host).getInteger("s3.assumerole.durationseconds")` which implements the fallback.

1. Search in properties saved with bookmark
2. Search in properties in connection profile
3. Search in application preferences

Reply:
This is now implemented according to your feedback.